docs/api/oauth2/README.md

```diff
@@ -84,19 +84,24 @@ Parameters:
 - `client_id`: Your application's Client ID
 - `scope`: Space-separated list of permissions your app needs
 - `redirect_uri`: Must match one of the URIs you registered
-- `state`: Random string to prevent CSRF attacks
+- `state`: Random string for security validation (see security note below)
 
-**Note**: If you provided an App connection URL, you have two options for the `redirect_uri`:
+**Note**: If you provided an App connection URL, you have two options for this URL:
 
 1. Set it directly to Orbitar's authorization endpoint with the parameters above
-   (`state` would have to be hardcoded)
-2. Set it to a URL on your app that will then redirect users to Orbitar's authorization endpoint
+   (you would need to hardcode the `state` parameter)
+2. Set it to a URL in your app that will then redirect users to Orbitar's authorization endpoint
    (recommended)
 
-The second option gives you more control and is more secure.
+The second option is more secure and gives you more control.
 
-**Security Note about State Parameter**: The `state` parameter is crucial for preventing Cross-Site Request Forgery (
-CSRF) attacks. Your application must:
+**Security Note about State Parameter**: The `state` parameter serves multiple security purposes:
+
+1. **Preventing CSRF attacks**: It ensures the authorization request came from your application
+2. **Session binding**: It binds the authorization request to the user's session
+3. **Protection against replay attacks**: It prevents authorization codes from being reused
```

Comment:
how?

Reply:
ah, I see, it's a reverse replay attack: when the callback is used multiple times.

needs to implemented on the thirdparty client side